### PR TITLE
Changes for character update functionality

### PR DESCRIPTION
**Pull request by ksmolko**
Front-end & Back-end code for handling character update functionality.

### CODE REVIEW

File: server/routes.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func (app *application) registerRoutes(e *echo.Echo) {
 	r.POST("/character", app.createCharacter)
 	r.GET("/character/me", app.retrieveUserCharacters)
 	r.GET("/character/:id", app.retrieveCharacter)
+	r.POST("/character/:id", app.updateCharacter)
```
Comment (knsjerome) [inline server/routes.go:29]: Generally HTTP POST is used for creating a new resource and HTTP PUT is used for updating, but at this point it doesn't matter.

Comment (ksmolko) [reply server/routes.go:29]: Ah right, will toss that in with the front-end change coming up.

File: client/src/components/characters/updating/CharacterInfo.vue [hunk 2]
```diff
@@ -41,10 +41,19 @@ characters. Enables other actions such as deleting characters. -->
...
+          <v-btn class="error" @click="displayCharacterDeletionPrompt">
+            Delete character
+            <v-icon>mdi-delete-forever</v-icon>
+          </v-btn>
+          <v-btn
+            class="ml-4"
+            color="primary"
+            @click="requestCharacterUpdate"
+          >
+            Update
```
Comment (knsjerome) [inline client/src/components/characters/updating/CharacterInfo.vue:54]: Maybe change to update character for consistency

File: client/src/components/characters/updating/CharacterInfo.vue [hunk 3]
```diff
@@ -342,6 +365,46 @@ export default {
...
+          hp_max: this.selectedCharacter.hp_max,
+          ability_points: this.selectedCharacter.ability_points,
+          xp_points: this.selectedCharacter.xp_points,
+          class: this.selectedCharacter.class,
+          class_attribute: this.selectedCharacter.class_attribute,
+          player_username: this.selectedCharacter.player_username,
+        },
+        method,
+      })
+        .then(() => {
```
Comment (knsjerome) [inline client/src/components/characters/updating/CharacterInfo.vue:400]: may wish to handle errors as well using a `.catch` block. You can copy-paste from other components

**Review by knsjerome — CHANGES_REQUESTED** [review]
Approved once changes are made

File: client/src/components/characters/updating/CharacterInfo.vue (hunk 2 above)
Comment (knsjerome) [reply client/src/components/characters/updating/CharacterInfo.vue:54]: Also I would move delete to be on the right so it's the same order of buttons as is used elsewhere (such as the campaigns page).

**Review by knsjerome — APPROVED** [review]

**Pull request merged**